api/core/v1alpha1/common_helpers.go

```diff
@@ -190,3 +202,64 @@ func findMainContainer(containers []corev1.Container) *corev1.Container {
 	}
 	return nil
 }
+
+// DefaultArgs contain default service args for logservice/dn/tp, these default args set in matrixone-operator-cm configmap
+type DefaultArgs struct {
+	LogService []string `json:"logService,omitempty"`
+	Dn         []string `json:"dn,omitempty"`
+	Tp         []string `json:"tp,omitempty"`
+}
+
+// loadDefaultArgsCfg read "defaultArgs" field in matrixone configmap, it is a single file named "defaultArgs"
+func loadDefaultArgsCfg(path string) (*DefaultArgs, error) {
+	bs, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	cfg := &DefaultArgs{}
+	err = yaml.Unmarshal(bs, cfg)
+	return cfg, err
+}
+
+// setDefaultServiceArgs set default args for service, we only set default args when there is service args config in service spec
+func setDefaultServiceArgs(object interface{}) {
+	cfgFile := filepath.Join(OperatorCfgPath, defaultArgsFile)
+	args, err := loadDefaultArgsCfg(cfgFile)
+	if err != nil {
+		moLog.Error(err, "load mo operator configmap fail")
+		return
+	}
+
+	switch obj := object.(type) {
+	case *LogSetBasic:
+		// set default arguments only when user does not set any arguments
+		if len(obj.ServiceArgs) == 0 {
+			obj.ServiceArgs = args.LogService
+		}
+		obj.ServiceArgs = trimEmptyArgs(obj.ServiceArgs)
```

Comment:
IMO we should respect the user's intent and report an error in validation if there is an empty arg so that the configuration error can be reported in early stage